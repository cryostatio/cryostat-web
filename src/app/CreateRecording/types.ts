/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { EventTemplate, KeyValue } from '@app/Shared/Services/api.types';
import { ValidatedOptions } from '@patternfly/react-core';

export type EventTemplateIdentifier = Pick<EventTemplate, 'name' | 'type'>;

interface _FormBaseData {
  name: string;
  template?: EventTemplateIdentifier;
  labels: KeyValue[];
  continuous: boolean;
  archiveOnStop: boolean;
  restart: boolean;
  duration: number;
  durationUnit: number;
  skipDurationCheck?: boolean;
  maxAge: number;
  maxAgeUnit: number;
  maxSize: number;
  maxSizeUnit: number;
  toDisk: boolean;
}

interface _FormValidationData {
  nameValid: ValidatedOptions;
  labelsValid: ValidatedOptions;
  durationValid: ValidatedOptions;
}

export type CustomRecordingFormData = _FormBaseData & _FormValidationData;
