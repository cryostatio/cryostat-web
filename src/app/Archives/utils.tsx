/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { RecordingDirectory, Target } from '@app/Shared/Services/api.types';

export const includesDirectory = (arr: RecordingDirectory[], dir: RecordingDirectory): boolean => {
  return arr.some((t) => t.connectUrl === dir.connectUrl);
};

export const indexOfDirectory = (arr: RecordingDirectory[], dir: RecordingDirectory): number => {
  let index = -1;
  arr.forEach((d, idx) => {
    if (d.connectUrl === dir.connectUrl) {
      index = idx;
    }
  });
  return index;
};

export const getTargetFromDirectory = (dir: RecordingDirectory): Target => {
  return {
    connectUrl: dir.connectUrl,
    alias: dir.jvmId,
    labels: [],
    annotations: {
      cryostat: [],
      platform: [],
    },
  };
};
