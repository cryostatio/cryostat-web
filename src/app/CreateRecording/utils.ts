/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { TemplateType } from '@app/Shared/Services/api.types';
import { EventTemplateIdentifier } from './types';

export const RecordingNamePattern = /^[\w_]+$/;

export const DurationPattern = /^[1-9][0-9]*$/;

export const isRecordingNameValid = (name: string) => RecordingNamePattern.test(name);

export const isDurationValid = (duration: number) => DurationPattern.test(`${duration}`);

export const templateFromEventSpecifier = (eventSpecifier: string): EventTemplateIdentifier | undefined => {
  const regex = /^template=([a-zA-Z0-9]+)(?:,type=([a-zA-Z0-9]+))?$/im;
  if (eventSpecifier && regex.test(eventSpecifier)) {
    const parts = regex.exec(eventSpecifier);
    if (parts) {
      return {
        name: parts[1],
        type: parts[2] as TemplateType,
      };
    }
  }
  return undefined;
};
