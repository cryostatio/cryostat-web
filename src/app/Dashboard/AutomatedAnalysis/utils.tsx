/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { AnalysisResult, Evaluation } from '@app/Shared/Services/api.types';
import { Stack, StackItem, Label, Title, Text } from '@patternfly/react-core';
import _ from 'lodash';

export const transformAADescription = (result: AnalysisResult): JSX.Element => {
  const format = (s: Evaluation): JSX.Element => {
    if (typeof s === 'string') {
      return <Text>{s}</Text>;
    }
    if (Array.isArray(s)) {
      return (
        <Stack>
          {s.map((e) => (
            <StackItem key={e.setting}>
              <Title headingLevel={'h6'}>{e.setting}</Title>
              <Label>
                {e.name}={e.value}
              </Label>
            </StackItem>
          ))}
        </Stack>
      );
    }
    throw `Unrecognized item: ${s}`;
  };
  return (
    <div>
      {Object.entries(result.evaluation || {}).map(([k, v]) =>
        v && v.length ? (
          <div key={k}>
            <span>
              <Title headingLevel={'h5'}>{_.capitalize(k)}</Title>
              {format(result.evaluation[k])}
            </span>
            <br />
          </div>
        ) : (
          <div key={k}></div>
        ),
      )}
    </div>
  );
};
