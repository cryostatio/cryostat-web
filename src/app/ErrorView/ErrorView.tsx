/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Button, EmptyState, EmptyStateBody, EmptyStateIcon, Title, StackItem, Stack } from '@patternfly/react-core';
import { ExclamationCircleIcon } from '@patternfly/react-icons';
import * as React from 'react';

export const authFailMessage = 'Authentication failure';

export const missingSSLMessage = 'Bad Gateway';

export const isAuthFail = (message: string) => message === authFailMessage;
export interface ErrorViewProps {
  title: string | React.ReactNode;
  message: string | React.ReactNode;
  retryButtonMessage?: string;
  retry?: () => void;
}

export const ErrorView: React.FC<ErrorViewProps> = (props) => {
  return (
    <>
      <EmptyState>
        <EmptyStateIcon icon={ExclamationCircleIcon} color={'#a30000'} />
        <Title headingLevel="h4" size="lg">
          {props.title}
        </Title>
        <EmptyStateBody>
          <>
            <Stack>
              <StackItem>{props.message}</StackItem>
              {props.retry && (
                <StackItem>
                  <Button variant="link" onClick={props.retry}>
                    {props.retryButtonMessage || 'Retry'}
                  </Button>
                </StackItem>
              )}
            </Stack>
          </>
        </EmptyStateBody>
      </EmptyState>
    </>
  );
};
