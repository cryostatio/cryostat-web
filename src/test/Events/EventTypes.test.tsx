/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { authFailMessage } from '@app/ErrorView/types';
import { EventTypes } from '@app/Events/EventTypes';
import { EventType, Target } from '@app/Shared/Services/api.types';
import { ServiceContext, defaultServices, Services } from '@app/Shared/Services/Services';
import { TargetService } from '@app/Shared/Services/Target.service';
import { act as doAct, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { of, Subject } from 'rxjs';
import { render, renderSnapshot } from '../utils';

const mockConnectUrl = 'service:jmx:rmi://someUrl';
const mockTarget = {
  connectUrl: mockConnectUrl,
  alias: 'fooTarget',
  jvmId: 'foo',
  labels: [],
  annotations: { cryostat: [], platform: [] },
};

const mockEventType: EventType = {
  name: 'Some Event',
  typeId: 'org.some_eventId',
  description: 'Some Descriptions',
  category: ['Category 1', 'Category 2'],
  options: [{ some_key: { name: 'some_name', description: 'a_desc', defaultValue: 'some_value' } }],
};

jest.spyOn(defaultServices.api, 'doGet').mockReturnValue(of([mockEventType]));
jest.spyOn(defaultServices.target, 'target').mockReturnValue(of(mockTarget));
jest.spyOn(defaultServices.target, 'authFailure').mockReturnValue(of());

describe('<EventTypes />', () => {
  afterEach(cleanup);

  it('renders correctly', async () => {
    const tree = await renderSnapshot({
      routerConfigs: {
        routes: [
          {
            path: '/events',
            element: <EventTypes />,
          },
        ],
      },
    });
    expect(tree?.toJSON()).toMatchSnapshot();
  });

  it('should show error view if failing to retrieve event types', async () => {
    const subj = new Subject<void>();
    const mockTargetSvc = {
      target: () => of(mockTarget as Target),
      authFailure: () => subj.asObservable(),
    } as TargetService;
    const services: Services = {
      ...defaultServices,
      target: mockTargetSvc,
    };

    render({
      routerConfigs: {
        routes: [
          {
            path: '/events',
            element: <EventTypes />,
          },
        ],
      },
      providers: [{ kind: ServiceContext.Provider, instance: services }],
    });

    await doAct(async () => subj.next());

    const failTitle = screen.getByText('Error retrieving event types');
    expect(failTitle).toBeInTheDocument();
    expect(failTitle).toBeVisible();

    const authFailText = screen.getByText(authFailMessage);
    expect(authFailText).toBeInTheDocument();
    expect(authFailText).toBeVisible();

    const retryButton = screen.getByText('Retry');
    expect(retryButton).toBeInTheDocument();
    expect(retryButton).toBeVisible();
  });

  it('should shown empty state when table is empty', async () => {
    const { user } = render({
      routerConfigs: {
        routes: [
          {
            path: '/events',
            element: <EventTypes />,
          },
        ],
      },
    });

    const filterInput = screen.getByLabelText('Event filter');
    expect(filterInput).toBeInTheDocument();
    expect(filterInput).toBeVisible();

    await user.type(filterInput, 'someveryoddname');

    expect(screen.queryByText('Some Event')).not.toBeInTheDocument();

    const hintText = screen.getByText('No Event Types');
    expect(hintText).toBeInTheDocument();
    expect(hintText).toBeVisible();
  });
});
