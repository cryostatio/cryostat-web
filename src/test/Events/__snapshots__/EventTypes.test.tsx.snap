// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventTypes /> renders correctly 1`] = `
Array [
  <div
    className="pf-v5-c-toolbar"
    data-ouia-component-id="OUIA-Generated-Toolbar-3"
    data-ouia-component-type="PF5/Toolbar"
    data-ouia-safe={true}
    id="event-types-toolbar"
  >
    <div
      className="pf-v5-c-toolbar__content"
    >
      <div
        className="pf-v5-c-toolbar__content-section"
      >
        <div
          className="pf-v5-c-toolbar__item"
        >
          <div
            className="pf-v5-c-text-input-group"
            id="eventFilter"
            style={
              Object {
                "minWidth": "38ch",
              }
            }
            type="search"
          >
            <div
              className="pf-v5-c-text-input-group__main pf-m-icon"
              onKeyDown={[Function]}
            >
              <span
                className="pf-v5-c-text-input-group__text"
              >
                <span
                  className="pf-v5-c-text-input-group__icon"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    className="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                    />
                  </svg>
                </span>
                <input
                  aria-label="event-type-search-input"
                  className="pf-v5-c-text-input-group__text-input"
                  disabled={false}
                  name="eventFilter"
                  onChange={[Function]}
                  placeholder="Find by name, description, typeId, or description..."
                  type="text"
                  value=""
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="pf-v5-c-toolbar__item pf-m-pagination"
        >
          <div
            className="pf-v5-c-pagination"
            data-ouia-component-id="OUIA-Generated-Pagination-top-3"
            data-ouia-component-type="PF5/Pagination"
            data-ouia-safe={true}
            id="event-types-pagination-top-pagination"
          >
            <div
              className="pf-v5-c-pagination__total-items"
            >
              <b>
                1
                 - 
                1
              </b>
               
              of
               
              <b>
                1
              </b>
               
              
            </div>
            <button
              aria-expanded={false}
              aria-haspopup="listbox"
              className="pf-v5-c-menu-toggle pf-m-plain pf-m-text"
              data-ouia-component-id="OUIA-Generated-MenuToggle-plainText-3"
              data-ouia-component-type="PF5/MenuToggle"
              data-ouia-safe={true}
              disabled={false}
              id="event-types-pagination-top-toggle"
              onClick={[Function]}
              type="button"
            >
              <span
                className="pf-v5-c-menu-toggle__text"
              >
                <b>
                  1
                   - 
                  1
                </b>
                 
                of
                 
                <b>
                  1
                </b>
                 
                
              </span>
              <span
                className="pf-v5-c-menu-toggle__controls"
              >
                <span
                  className="pf-v5-c-menu-toggle__toggle-icon"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    className="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 320 512"
                    width="1em"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <nav
              aria-label="Pagination"
              className="pf-v5-c-pagination__nav"
            >
              <div
                className="pf-v5-c-pagination__nav-control pf-m-first"
              >
                <button
                  aria-disabled={true}
                  aria-label="Go to first page"
                  className="pf-v5-c-button pf-m-plain pf-m-disabled"
                  data-action="first"
                  data-ouia-component-id="OUIA-Generated-Button-plain-9"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe={true}
                  disabled={true}
                  onClick={[Function]}
                  role={null}
                  tabIndex={null}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    className="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 448 512"
                    width="1em"
                  >
                    <path
                      d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                    />
                  </svg>
                </button>
              </div>
              <div
                className="pf-v5-c-pagination__nav-control"
              >
                <button
                  aria-disabled={true}
                  aria-label="Go to previous page"
                  className="pf-v5-c-button pf-m-plain pf-m-disabled"
                  data-action="previous"
                  data-ouia-component-id="OUIA-Generated-Button-plain-10"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe={true}
                  disabled={true}
                  onClick={[Function]}
                  role={null}
                  tabIndex={null}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    className="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                    />
                  </svg>
                </button>
              </div>
              <div
                className="pf-v5-c-pagination__nav-page-select"
              >
                <span
                  className="pf-v5-c-form-control pf-m-disabled"
                >
                  <input
                    aria-invalid={false}
                    aria-label="Current page"
                    data-ouia-component-id="OUIA-Generated-TextInputBase-3"
                    data-ouia-component-type="PF5/TextInput"
                    data-ouia-safe={true}
                    disabled={true}
                    max={1}
                    min={1}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    required={false}
                    type="number"
                    value={1}
                  />
                </span>
                <span
                  aria-hidden="true"
                >
                  of
                   
                  1
                </span>
              </div>
              <div
                className="pf-v5-c-pagination__nav-control"
              >
                <button
                  aria-disabled={true}
                  aria-label="Go to next page"
                  className="pf-v5-c-button pf-m-plain pf-m-disabled"
                  data-action="next"
                  data-ouia-component-id="OUIA-Generated-Button-plain-11"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe={true}
                  disabled={true}
                  onClick={[Function]}
                  role={null}
                  tabIndex={null}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    className="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </button>
              </div>
              <div
                className="pf-v5-c-pagination__nav-control pf-m-last"
              >
                <button
                  aria-disabled={true}
                  aria-label="Go to last page"
                  className="pf-v5-c-button pf-m-plain pf-m-disabled"
                  data-action="last"
                  data-ouia-component-id="OUIA-Generated-Button-plain-12"
                  data-ouia-component-type="PF5/Button"
                  data-ouia-safe={true}
                  disabled={true}
                  onClick={[Function]}
                  role={null}
                  tabIndex={null}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    className="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 448 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                    />
                  </svg>
                </button>
              </div>
            </nav>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pf-v5-c-toolbar__content pf-m-hidden"
      hidden={true}
    >
      <div
        className="pf-v5-c-toolbar__group"
      />
    </div>
  </div>,
  <table
    aria-label="Event types Table"
    className="pf-v5-c-table pf-m-grid-md pf-m-compact"
    data-ouia-component-id="OUIA-Generated-Table-3"
    data-ouia-component-type="PF5/Table"
    data-ouia-safe={true}
    role="grid"
  >
    <thead
      className="pf-v5-c-table__thead"
    >
      <tr
        className="pf-v5-c-table__tr"
        data-ouia-component-id="OUIA-Generated-TableRow-7"
        data-ouia-component-type="PF5/TableRow"
        data-ouia-safe={true}
        hidden={false}
      >
        <th
          className="pf-v5-c-table__th"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          scope="col"
          tabIndex={-1}
        />
        <th
          aria-sort="none"
          className="pf-v5-c-table__th pf-v5-c-table__sort"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          scope="col"
          tabIndex={-1}
        >
          <button
            className="pf-v5-c-table__button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            type="button"
          >
            <div
              className="pf-v5-c-table__button-content"
            >
              <span
                className="pf-v5-c-table__text"
                onMouseEnter={[Function]}
              >
                Name
              </span>
              <span
                className="pf-v5-c-table__sort-indicator"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                  />
                </svg>
              </span>
            </div>
          </button>
        </th>
        <th
          aria-sort="none"
          className="pf-v5-c-table__th pf-v5-c-table__sort"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          scope="col"
          tabIndex={-1}
        >
          <button
            className="pf-v5-c-table__button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            type="button"
          >
            <div
              className="pf-v5-c-table__button-content"
            >
              <span
                className="pf-v5-c-table__text"
                onMouseEnter={[Function]}
              >
                Type ID
              </span>
              <span
                className="pf-v5-c-table__sort-indicator"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                  />
                </svg>
              </span>
            </div>
          </button>
        </th>
        <th
          aria-sort="none"
          className="pf-v5-c-table__th pf-v5-c-table__sort"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          scope="col"
          tabIndex={-1}
        >
          <button
            className="pf-v5-c-table__button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            type="button"
          >
            <div
              className="pf-v5-c-table__button-content"
            >
              <span
                className="pf-v5-c-table__text"
                onMouseEnter={[Function]}
              >
                Description
              </span>
              <span
                className="pf-v5-c-table__sort-indicator"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                  />
                </svg>
              </span>
            </div>
          </button>
        </th>
        <th
          aria-sort="none"
          className="pf-v5-c-table__th pf-v5-c-table__sort"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          scope="col"
          tabIndex={-1}
        >
          <button
            className="pf-v5-c-table__button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            type="button"
          >
            <div
              className="pf-v5-c-table__button-content"
            >
              <span
                className="pf-v5-c-table__text"
                onMouseEnter={[Function]}
              >
                Categories
              </span>
              <span
                className="pf-v5-c-table__sort-indicator"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                  />
                </svg>
              </span>
            </div>
          </button>
        </th>
      </tr>
    </thead>
    <tbody
      className="pf-v5-c-table__tbody"
      role="rowgroup"
    >
      <tr
        className="pf-v5-c-table__tr"
        data-ouia-component-id="OUIA-Generated-TableRow-8"
        data-ouia-component-type="PF5/TableRow"
        data-ouia-safe={true}
        hidden={false}
      >
        <td
          className="pf-v5-c-table__td pf-v5-c-table__toggle"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          tabIndex={-1}
        >
          <button
            aria-disabled={false}
            aria-expanded={false}
            aria-label="Details"
            aria-labelledby="simple-node0 expandable-event-type-row-00"
            className="pf-v5-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-3"
            data-ouia-component-type="PF5/Button"
            data-ouia-safe={true}
            disabled={false}
            id="expandable-event-type-row-00"
            onClick={[Function]}
            role={null}
            type="button"
          >
            <div
              className="pf-v5-c-table__toggle-icon"
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                className="pf-v5-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                />
              </svg>
            </div>
          </button>
        </td>
        <td
          className="pf-v5-c-table__td"
          data-label="Name"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          tabIndex={-1}
        >
          Some Event
        </td>
        <td
          className="pf-v5-c-table__td"
          data-label="Type ID"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          tabIndex={-1}
        >
          org.some_eventId
        </td>
        <td
          className="pf-v5-c-table__td"
          data-label="Description"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          tabIndex={-1}
        >
          Some Descriptions
        </td>
        <td
          className="pf-v5-c-table__td"
          data-label="Categories"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          tabIndex={-1}
        >
          Category 1, Category 2
        </td>
      </tr>
      <tr
        className="pf-v5-c-table__tr pf-v5-c-table__expandable-row"
        data-ouia-component-id="OUIA-Generated-TableRow-9"
        data-ouia-component-type="PF5/TableRow"
        data-ouia-safe={true}
        hidden={true}
      >
        <td
          className="pf-v5-c-table__td"
          colSpan={4}
          data-label="event-details"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          tabIndex={-1}
        >
          <div
            className="pf-v5-c-table__expandable-row-content"
          >
            <p
              className=""
              data-ouia-component-id="OUIA-Generated-Text-3"
              data-ouia-component-type="PF5/Text"
              data-ouia-safe={true}
              data-pf-content={true}
            >
              0=[undefined]	
            </p>
          </div>
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
